main.tf

```diff
@@ -199,3 +199,49 @@ module "self_serve_access_keys" {
 
   user_names = aws_iam_user.elasticsearch_iam_user.*.name
 }
+
+
+resource "aws_iam_user" "elasticsearch_iam_users" {
+  for_each = var.iam_users
+  name     = "${var.name}-${each.key}-iam-user"
+
+  tags = merge(
+    var.tags,
+    local.email_tags,
+    {
+      "key_rotation" = var.key_rotation
+    },
+  )
+}
+
+data "aws_iam_policy_document" "elasticsearch_iam_users_policy" {
+  for_each = var.iam_users
+
+  dynamic "statement" {
+    for_each = each.value
+
+    content {
+      effect = "Allow"
```

Comment:
Do you think we should restrict the actions to **ESHttp*** only as the potential actions seems quite wide: https://aws.permissions.cloud/iam/es